docs/api/connection.rst

```diff
@@ -0,0 +1,278 @@
+
+        .. code-block:: js
+
+            await con.execute(`
+                CREATE TYPE MyType {
+                    CREATE PROPERTY a -> int64
+                };
+                FOR x IN {100, 200, 300}
+                UNION INSERT MyType { a := x };
+            `)
+
+    .. js:method:: fetchAll(query: string, args)
+
+        Run a query and return the results as a
+        :js:class:`Set` instance.
+
+        This method takes :js:class:`optional arguments <AwaitConnection>`.
+
+    .. js:method:: fetchOne(query: string, args)
+
+        Run a singleton-returning query and return its element.
+
+        This method takes :js:class:`optional arguments <AwaitConnection>`.
+
+        The *query* must return exactly one element.  If the query returns
+        more than one element or an empty set, an ``Error`` is thown.
+
+    .. js:method:: fetchAllJSON(query: string, args)
+
+        Run a query and return the results as JSON.
+
+        This method takes :js:class:`optional arguments <AwaitConnection>`.
+
+    .. js:method:: fetchOneJSON(query: string, args)
+
+        Run a singleton-returning query and return its element in JSON.
+
+        This method takes :js:class:`optional arguments <AwaitConnection>`.
+
+        The *query* must return exactly one element.  If the query returns
+        more than one element or an empty set, an ``Error`` is thown.
+
+    .. js:method:: close()
+
+        Close the connection gracefully.
+
+.. js:class:: Connection
+
+    A representation of a database session.
+
+    :js:class:`Connection` is not meant to be instantiated by directly,
+    :js:func:`connect` should be used instead.
+
+    Every method of this class takes a *callback* of the form
+    ``function(err, data)``.
+
+    Some methods take query arguments as optional *args*:
+
+    * single values of any of the :ref:`basic types
+      recognized<edgedb-js-datatypes>` by EdgeDB
+    * an ``Array`` of values of any of the basic types
+    * an ``object`` with property names and values corresponding to
+      argument names and values of any of the basic types
+
+    .. js:method:: execute(query: string, callback)
+
+        Execute an EdgeQL command (or commands).
+
+        The commands must take no arguments.
+
+        Example:
+
+        .. code-block:: js
+
+            con.execute(`
+                CREATE TYPE MyType {
+                    CREATE PROPERTY a -> int64
+                };
+                FOR x IN {100, 200, 300}
+                UNION INSERT MyType { a := x };
+            `, (err, data) => {
+                if (err) {
+                    console.log('migration failed: ', err)
+                } else {
+                    console.log('migration complete');
+                }
+            })
+
+    .. js:method:: fetchAll(query: string, args)
+
+        Run a query and return the results as a
+        :js:class:`Set` instance.
+
+        This method takes :js:class:`optional arguments <Connection>`.
```

Comment:
`optional arguments` -- you mean "optional query arguments"?

Reply:
Yes. Updated.